### ISSUE TITLE
gem command neglects --pre option

### ISSUE THREAD

**Issue by igaiga** [rubygems/rubygems#4112]
I'm having a problem or would like to suggest a feature.

My current problem is "gem command neglects --pre option".

---
I've run `gem i rails --pre` command on Ruby 3.0.0preview2 and gem v.3.2.0, gem command installed rails-6.0.3.4 (not rails 6.1.0.rc2).
It looks that gem command neglect --pre option.

```
$ gem i rails --pre
Fetching rails-6.0.3.4.gem
Successfully installed rails-6.0.3.4
1 gem installed

$ ruby -v
ruby 3.0.0preview2 (2020-12-08 master d7a16670c3) [x86_64-darwin19]

$ gem -v
3.2.0

$ gem update --system
Latest version already installed. Done.
```

- Ruby: 3.0.0preview2
  - rbenv, ruby-build
- gem: 3.2.0
- OS: MacOS 10.15.7

In Ruby 2.7.2 and gem v.3.1.4, gem command installs rails 6.1.0.rc2 as intended.

---

This issue is related to:

  - [ ] Network problems
  - [x] Installing a library
  - [ ] Publishing a library
  - [x] The command line `gem`
  - [ ] Other

Here are my current environment details:

```
$ gem env version
3.2.0
```

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

**Comment by MSP-Greg** [comment]
Maybe `--prerelease` ?

**Comment by igaiga** [comment]
`gem install rails --prerelease` didn't work. 😢
It works same as `--pre` option.

**Comment by deivid-rodriguez** [comment]
I can reproduce this issue. This is an unintended regression from the upgrade to molinillo 0.7.0 in #3388. I will have a look and revert the upgrade if I can't figure it out.

**Comment by deivid-rodriguez** [comment]
@igaiga #4113 should fix this in case you want to give it a try.

**Comment by igaiga** [comment]
@deivid-rodriguez Thanks! 😃 
I've tryed your PR, but rails gem is released 6.1.0 final version a few hours ago, so I couldn't try the case to work --pre option...
